### PR TITLE
Fixed CSV file upload does not correctly handle 'UTF-8 with BOM' encoding

### LINKED ISSUE #5812: CSV file upload doesn't correctly handle "UTF-8 with BOM" CSVs
### Environment
* Nautobot version (Docker tag too if applicable): v2.2.25
* Python version: 3.11
* Database platform, version: PG15
* Middleware(s): N/A

### Steps to Reproduce
1. Affecting [ipam/prefixes](https://demo.nautobot.com/ipam/prefixes/) CSV Import feature
2. Upload a bulk Prefix CSV file to `Prefix` that contains the columns marked as required: prefix, status
image
4. The job will get an error: `Row <id>: prefix : this field is required.`
image

### Expected Behavior
Upload is parsed without error, job completes

### Observed Behavior
job error: `Row <id>: prefix : this field is required.`

### Mitigation
The unspecified column `display` is included in the CSV export (including the `prefix` column separately)
For the CSV Upload to complete I mirrored this, and include both a `display` and `prefix` colum in my CSV upload

### Headers from Prefix CSV Export
`display,id,object_type,natural_slug,prefix,type,network,broadcast,prefix_length,ip_version,date_allocated,description,status__name,role__name,parent__namespace__name,parent__network,parent__prefix_length,namespace__name,tenant__name,vlan__pk,rir__name,created,last_updated,tags`

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes #5812, #5985
# What's Changed

Used a fresh .csv file and the failing .csv file in the issue description to test this fix and they both succeeded.
# Screenshots

N/A
# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/core/jobs/__init__.py [hunk 1]
```diff
@@ -282,7 +283,7 @@ def run(self, *, content_type, csv_data=None, csv_file=None, roll_back_if_error=
         if not csv_data and not csv_file:
             raise RunJobTaskFailed("Either csv_data or csv_file must be provided")
         if csv_file:
-            csv_bytes = csv_file
+            csv_bytes = codecs.EncodedFile(csv_file, "utf-8", "utf-8-sig")
```
Comment (HanlinMiao) [inline nautobot/core/jobs/__init__.py:286]: utf-8 is the data encoding and utf-8-sig is the file encoding. "Bytes read from the original file are decoded according to file_encoding, and the result is encoded using data_encoding." From the python [doc](https://docs.python.org/3/library/codecs.html#codecs.EncodedFile)

**Review by glennmatthews — COMMENTED** [review]
Can we add a unit test, possibly using one of the simple files provided in the issue?

**Comment by glennmatthews** [comment]
Let's confirm that this also fixes #5985 as well.

File: nautobot/core/jobs/__init__.py (hunk 1 above)
Comment (glennmatthews) [reply nautobot/core/jobs/__init__.py:286]: Maybe add as a comment in the code?

Comment (timizuoebideri1) [reply nautobot/core/jobs/__init__.py:288]: Do we only want a  fix for `utf-8-sig` are there not other encoding we might want to consider?
or better still only accept `UTF-8` and throw an error when encoding is not in `utf-8`

Comment (HanlinMiao) [reply nautobot/core/jobs/__init__.py:288]: I think the current solution takes utf-8 and utf-8-sig which is more desirable than utf-8 alone. From the issue description, it seems like a lot of the users are using utf-8 with BOM, whether advertently or inadvertently. So IMO, we should support both.

**Review by timizuoebideri1 — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**